spec/node/asciidoctor.spec.js

```diff
@@ -109,6 +109,42 @@ intro
           asciidoctor.LoggerManager.setLogger(defaultLogger);
         }
       });
+      it('should create a custom Logger', (done) => {
+        const input = `= Book
+:doctype: book
+
+= Part 1
+
+[partintro]
+intro
+`;
+        const fs = require('fs');
+        const defaultLogger = asciidoctor.LoggerManager.getLogger();
+        const logFile = path.join(__dirname, '..', '..', 'build', 'async.log');
+        const asyncLogger = asciidoctor.LoggerManager.newLogger('AsyncFileLogger', {
+          postConstruct: function () {
+            this.writer = fs.createWriteStream(logFile, {
+              flags: 'a'
+            });
+            fs.truncateSync(logFile, 0); // file must be empty
+          },
+          add: function (severity, _, message) {
+            const log = this.formatter['$call'](asciidoctor.LoggerSeverity.get(severity), new Date(), this.progname, message);
```

Comment:
Will be replaced with a proper API call once https://github.com/asciidoctor/asciidoctor.js/pull/553 is merged